src/scripts/outlet-registration/index.js

```diff
@@ -0,0 +1,582 @@
+            console.error("\t",event.orgUnit," => cannot resolve 'parentCode'");
+            _this.nextEvent();
+            return;    
+        }
+        
+        //resolve autoinc
+        event.autoIncrement = _this.findLastAutoIncrement(event,organisationUnits) +1;
+        
+        if(!event.autoIncrement){
+            console.error("\t",event.orgUnit," => cannot resolve 'autoIncrement'");
+            _this.nextEvent();
+            return;                
+        }
+               
+        //Post orgunit 
+        _this.postOrgUnit(event);
+    });    
+};
+
+/**
+ * Returns the orgUnit.code from the given event.orgUnit
+ * @param event The event 
+ */
+OutletRegistrator.prototype.findParentCode = function(event,organisationUnits) {    
+    var organisationUnit = organisationUnits.find(organisationUnit =>{
+       return organisationUnit.id === event.orgUnit 
+    });
+    
+    return organisationUnit?organisationUnit.code.split("_")[1]:null;
+};
+
+/**
+ * Returns the orgUnit.organisationUnits.'maxcode' from the given event.orgUnit
+ * @param event The event 
+ */
+OutletRegistrator.prototype.findLastAutoIncrement = function(event,organisationUnits) { 
+    var max = 0;
+    
+	//filter only those following the right sequence 'MM_AMTR[PARENTCODE]-'
+    var onlyAMTROrgUnits = organisationUnits.filter(organisationUnit => organisationUnit.code.indexOf("MM_AMTR"+event.parentCode+"-")!==-1);
+    onlyAMTROrgUnits.forEach(function(AMTROrgUnit){
+        var currentValue = parseInt(AMTROrgUnit.code.split("-")[1]);
+        if (!isNaN(currentValue)) {
+    	    max = currentValue>max?currentValue:max;
+        }    
+    });
+
+    return max;    
+};
+
+/**
+ * Post the new OrgUnit and PATCHES the event (alreadyImported:true)
+ * @param event The event that will be converted into an orgunit
+ */
+OutletRegistrator.prototype.postOrgUnit = function(event) {    
+    console.log("\t parentCode: ",event.parentCode," autoIncrement: "+event.autoIncrement);
+    //Prepare orgUnit
+    var newOrgUnit = this.createOrgUnitFromEvent(event);
+    //Post orgunit  
+    if (!this.ifValidOrgUnit(newOrgUnit)) {
+        console.log("Compulsory fields have not been filled ",newOrgUnit);
+        this.nextEvent();
+        return
+    }
+    this.postAndPatch(newOrgUnit, event);
+  
+};
+
+/**
+ * Returns a string with outlet code
+ * {AMTR}{ParentCode}{-}{Increment}
+ * @params parentCode of the parent org. unit
+ * @params autoIncrement integer to be added to the code
+ */
+OutletRegistrator.prototype.createOrgUnitCode = function(parentCode,autoIncrement) {
+    var formatNumber = (autoIncrement>0 && autoIncrement<10)?"0"+autoIncrement:autoIncrement;
+    return "AMTR"+parentCode+"-"+formatNumber;
+};
+
+/**
+ * DHIS2 format for coordinates [longitude,latitude]
+ * @params coord [longitude, latitude] coordinates
+ */
+OutletRegistrator.prototype.setupCoordiantes = function(coord) {
+    coordinates = [coord.longitude, coord.latitude];
```

Comment:
This variable is not needed, just return the array. Something for you to keep in mind, if you define a variable without 'var' in js the variable scope will be global. This is a bit dangerous IMO and I would not recommend to do it unless there is a good reason. In this case I don't think it is used at a global level but correct me if I am wrong